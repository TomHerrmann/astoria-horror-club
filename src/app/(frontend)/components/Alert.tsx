import { Alert as BaseAlert, AlertDescription, AlertTitle } from '@/components/ui/alert'
import { CircleX, CircleCheck, CircleAlert } from 'lucide-react'

export type Props = {
  title: string
  description: string
  variant?: 'success' | 'destructive' | 'nuetral'
  styles?: string
}

function AlertIcon({ variant }: Pick<Props, 'variant'>) {
  switch (variant) {
    case 'destructive':
      return <CircleX className="h-4 w-4" />
    case 'success':
      return <CircleCheck className="h-4 w-4" />
    default:
      return <CircleAlert className="h-4 w-4" />
  }
}

export default function Alert({ description, title, variant, styles }: Props) {
  return (
    <BaseAlert variant={variant === 'destructive' ? 'destructive' : 'default'} className={styles}>
      <AlertIcon variant={variant} />
      <AlertTitle>{title}</AlertTitle>
      <AlertDescription>{description}</AlertDescription>
    </BaseAlert>
  )
}
