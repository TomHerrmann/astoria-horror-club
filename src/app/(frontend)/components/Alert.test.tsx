import { describe, it, expect, vi } from 'vitest'
import { render, screen } from '@testing-library/react'
import Alert from './Alert'

// Mock Lucide icons
vi.mock('lucide-react', () => ({
  CircleX: () => <div data-testid="circle-x-icon" />,
  CircleCheck: () => <div data-testid="circle-check-icon" />,
  CircleAlert: () => <div data-testid="circle-alert-icon" />,
}))

// Mock ShadCN components
vi.mock('@/components/ui/alert', () => ({
  Alert: ({ children, variant, className }: any) => (
    <div data-testid="base-alert" data-variant={variant} className={className}>
      {children}
    </div>
  ),
  AlertTitle: ({ children }: any) => <div data-testid="alert-title">{children}</div>,
  AlertDescription: ({ children }: any) => <div data-testid="alert-description">{children}</div>,
}))

describe('Alert Component', () => {
  const testProps = {
    title: 'Test Title',
    description: 'Test description message',
    variant: 'nuetral' as const,
  }

  describe('AlertIcon', () => {
    it('matches snapshot for each variant', () => {
      const { container: nuetral } = render(<Alert {...testProps} />)
      expect(nuetral).toMatchSnapshot()

      const { container: destructive } = render(<Alert {...testProps} variant="destructive" />)
      expect(destructive).toMatchSnapshot()

      const { container: success } = render(<Alert {...testProps} variant="success" />)
      expect(success).toMatchSnapshot()
    })

    it('renders CircleX for destructive variant', () => {
      render(<Alert {...testProps} variant="destructive" />)
      expect(screen.getByTestId('circle-x-icon')).toBeDefined()
    })

    it('renders CircleCheck for success variant', () => {
      render(<Alert {...testProps} variant="success" />)
      expect(screen.getByTestId('circle-check-icon')).toBeDefined()
    })

    it('renders CircleAlert for nuetral variant', () => {
      render(<Alert {...testProps} variant="nuetral" />)
      expect(screen.getByTestId('circle-alert-icon')).toBeDefined()
    })

    it('renders CircleAlert for default variant', () => {
      render(<Alert {...testProps} variant={undefined} />)
      expect(screen.getByTestId('circle-alert-icon')).toBeDefined()
    })
  })

  describe('Alert', () => {
    it('renders with default props', () => {
      render(<Alert {...testProps} />)

      const baseAlert = screen.getByTestId('base-alert')
      expect(baseAlert).toBeDefined()
      expect(baseAlert.dataset.variant).toBe('default')

      expect(screen.getByText('Test Title')).toBeDefined()
      expect(screen.getByText('Test description message')).toBeDefined()
      expect(screen.getByTestId('circle-alert-icon')).toBeDefined()
    })

    it('applies destructive variant correctly', () => {
      render(<Alert {...testProps} variant="destructive" />)

      expect(screen.getByTestId('circle-x-icon')).toBeDefined()
    })

    it('applies success variant correctly', () => {
      render(<Alert {...testProps} variant="success" />)

      expect(screen.getByTestId('circle-check-icon')).toBeDefined()
    })
  })
})
