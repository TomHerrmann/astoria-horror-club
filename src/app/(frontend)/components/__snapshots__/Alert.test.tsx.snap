// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Alert Component > AlertIcon > matches snapshot for each variant 1`] = `
<div>
  <div
    data-testid="base-alert"
    data-variant="default"
  >
    <div
      data-testid="circle-alert-icon"
    />
    <div
      data-testid="alert-title"
    >
      Test Title
    </div>
    <div
      data-testid="alert-description"
    >
      Test description message
    </div>
  </div>
</div>
`;

exports[`Alert Component > AlertIcon > matches snapshot for each variant 2`] = `
<div>
  <div
    data-testid="base-alert"
    data-variant="destructive"
  >
    <div
      data-testid="circle-x-icon"
    />
    <div
      data-testid="alert-title"
    >
      Test Title
    </div>
    <div
      data-testid="alert-description"
    >
      Test description message
    </div>
  </div>
</div>
`;

exports[`Alert Component > AlertIcon > matches snapshot for each variant 3`] = `
<div>
  <div
    data-testid="base-alert"
    data-variant="default"
  >
    <div
      data-testid="circle-check-icon"
    />
    <div
      data-testid="alert-title"
    >
      Test Title
    </div>
    <div
      data-testid="alert-description"
    >
      Test description message
    </div>
  </div>
</div>
`;
